Add competitor comparison table (#1366)
meilisearch/documentation: learn/what_is_meilisearch/comparison_to_alternatives.md

| All language supports | ✅ | ✅ | ❌ <br> Only space separated | ✅ |

##### Security

|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
|---|:---:|:----:|:---:|:---:|
| API Key Management | ✅ | ✅ | ✅ | ✅ |
| Tenant tokens & multi-tenant indexes | **Q1 2022** <br> Configurable end-user hard filters per index | ✅  <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Role-based |

##### Search

|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
|---|:---:|:----:|:---:|:---:|
| Placeholder search | ✅ | ✅ | ✅ | ✅ |
| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
| Geo search |  ✅  | ✅ | ✅ | ✅ |
| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Indexes have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
| Filtering |  ✅ <br> Support complex filter queries with an SQL-like syntax.  | 🔶 <br> Does not support `OR` operation across multiple fields | 🔶 <br> Does not support `OR` operation across multiple fields | ✅ |

> | Faceting |  ✅ | ✅ | ✅ | ✅ |

Reviewer: ```suggestion
| Faceted search |  ✅ | ✅ | ✅ | ✅ |
```

> | Faceting |  ✅ | ✅ | ✅ | ✅ |

Author: ```suggestion
| Faceted search |  ✅ | ✅ | ✅ | ✅ |
```

- Finally, we end this article with [an in-depth analysis of the broader search engine landscape](#a-quick-look-at-the-search-engine-landscape)

::: note
Please be advised that many of the search products described below are constantly evolving—just like MeiliSearch. These are only our own impressions, and may not reflect recent changes. If something appears inaccurate, please don't hesitate to open an [issue or pull request](https://github.com/meilisearch/documentation).
:::

## Comparison table

### General Overview

|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
|---|:----:|:----:|:-----:|:----:|
| Source code licensing | [MIT](https://choosealicense.com/licenses/mit/) <br> (Fully open-source) |  Closed-source  | [GPL-3](https://choosealicense.com/licenses/gpl-3.0/) <br> (Fully open-source) | SSPL <br>([Not open-source](https://opensource.org/node/1099))  |
| Built with | Rust <br> [Check out why we believe in Rust](https://www.abetterinternet.org/docs/memory-safety/). | C++ | C++ | Java |
| Data storage | Disk with Memory Mapping -- Not limited by RAM | Limited by RAM | Limited by RAM | Disk with RAM cache |

### Features

#### Integrations and SDKs

> We list only officially supported libraries by the internal teams of the different projects.

Author: ```suggestion
Note: we are only listing libraries officially supported by the internal teams of each different search engine.
```

| Typo tolerant  | ✅ | ✅ | ✅ | 🔶 <br>Needs to be specified by fuzzy queries |
| Orderable ranking rules | ✅ | ✅ | 🔶 <br>Tie-breaking order is limited by a unique scoring rule | ❌|
| Custom rules | ✅ | ✅ | 🔶 <br>Limited to one default sorting rule | 🔶 <br>Function score query
| Query field weights | ✅ | ✅ | ✅ | ✅ |
| Synonyms | ✅ | ✅ | ✅ | ✅ |
| Stop words | ✅ | ✅ | ❌ | ✅ |
| Automatic language detection | ✅ | ✅ | ❌ | ❌ |
| All language supports | ✅ | ✅ | ❌ <br> Only space separated | ✅ |

##### Security

|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
|---|:---:|:----:|:---:|:---:|
| API Key Management | ✅ | ✅ | ✅ | ✅ |
| Tenant tokens & multi-tenant indexes | **Q1 2022** <br> Configurable end-user hard filters per index | ✅  <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Hard filters are not configurable per index for an end-user tenant key | ✅ <br> Role-based |

##### Search

|   | MeiliSearch | Algolia | Typesense | Elasticsearch |
|---|:---:|:----:|:---:|:---:|
| Placeholder search | ✅ | ✅ | ✅ | ✅ |
| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
| Geo search |  ✅  | ✅ | ✅ | ✅ |

> | Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Index have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |

Author: ```suggestion
| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Indexes have to be duplicated for each sort field and sort order | 🔶 <br> Does not support sort on string field | ✅ |
```

| Filtering |  ✅ <br> Support complex filter queries with an SQL-like syntax.  | 🔶 <br> Does not support `OR` operation across multiple fields | 🔶 <br> Does not support `OR` operation across multiple fields | ✅ |
| Faceting |  ✅ | ✅ | ✅ | ✅ |

> | Facet value searches | ❌ | ✅ | ✅ | ❌ |

Author: ```suggestion
```
